src/policy/rbf.cpp

```diff
@@ -161,3 +161,29 @@ std::optional<std::string> PaysMoreThanConflicts(const CTxMemPool::setEntries& s
     return std::nullopt;
 }
 
+std::optional<std::string> PaysForRBF(CAmount nConflictingFees,
+                                      CAmount nModifiedFees,
+                                      size_t nSize,
+                                      const uint256& hash)
+{
+    // The replacement must pay greater fees than the transactions it
+    // replaces - if we did the bandwidth used by those conflicting
+    // transactions would not be paid for.
+    if (nModifiedFees < nConflictingFees)
+    {
+        return strprintf("rejecting replacement %s, less fees than conflicting txs; %s < %s",
+                         hash.ToString(), FormatMoney(nModifiedFees), FormatMoney(nConflictingFees));
+    }
+
+    // Finally in addition to paying more fees than the conflicts the
+    // new transaction must pay for its own bandwidth.
+    CAmount nDeltaFees = nModifiedFees - nConflictingFees;
+    if (nDeltaFees < ::incrementalRelayFee.GetFee(nSize))
```

Comment:
ac761f0a23c9c469fa00885edf3d5c9ae7c6a2b3: Would be nice to not use a global here. Testability is one of the motivations in OP, but using globals makes testing actually harder.

Reply:
Yes definitely agree - I can do this in the followup PR.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/22855/commits/7ccaac6c5332b8a93c9b5c7103f846854ee595d5